File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0

def dfs(i, v, n, t):
    global cnt 
    if i == len(n):
        if v == t:
            cnt+= 1
        return 
    
    dfs(i+1, v + n[i], n, t)
    dfs(i+1, v - n[i], n, t)


def solution(numbers, target):
    global cnt
    dfs(0,0, numbers, target)
    
    return cnt

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public static int tarNum;
    public static int answerCnt = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        int max_cnt = numbers.length - 1;
        tarNum = target;

        dfs(max_cnt, -1, numbers, 0);

        answer = answerCnt;
        return answer;
    }

    public void dfs(int maxCnt, int depth, int[] numbers, int sum) {
        if (depth == maxCnt) {
            if (tarNum == sum) {
                answerCnt++;
            }
        } else {
            dfs(maxCnt, depth+1, numbers, sum + numbers[depth+1]);
            dfs(maxCnt, depth+1, numbers, sum - numbers[depth+1]);
        }
    }
}